common/logisticspipes/gui/guidebook/GuiGuideBookTexturedButton.java

```diff
@@ -0,0 +1,104 @@
+package logisticspipes.gui.guidebook;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.util.ResourceLocation;
+
+import com.sun.jna.platform.win32.WinUser;
+import lombok.Setter;
+
+import logisticspipes.LPConstants;
+
+public class GuiGuideBookTexturedButton extends GuiButton {
+
+	private int u0, v0, u1, v1, sizeX, sizeY;
+	@Setter
+	private boolean hasDisabledState;
+	@Setter
+	private EnumButtonType type;
+
+	private static final ResourceLocation GUI_BOOK_TEXTURE = new ResourceLocation(LPConstants.LP_MOD_ID, "textures/gui/guide_book.png");
+
+	public GuiGuideBookTexturedButton(int buttonId, int x, int y, int widthIn, int heightIn) {
+		super(buttonId, x, y, widthIn, heightIn, "");
+		this.u0 = 0;
+		this.v0 = 0;
+		this.u1 = 0;
+		this.v1 = 0;
+		this.sizeX = 0;
+		this.sizeY = 0;
+		this.hasDisabledState = false;
+		this.type = EnumButtonType.TAB;
+	}
+
+	public GuiGuideBookTexturedButton(int buttonId, int x, int y, int widthIn, int heightIn, int u0, int v0, int z, int u1, int v1, int sizeX, int sizeY, boolean hasDisabledState, EnumButtonType type){
+		super(buttonId, x, y, widthIn, heightIn, "");
+		this.u0 = u0;
+		this.v0 = v0;
+		this.u1 = u1;
+		this.v1 = v1;
+		this.zLevel = z;
+		this.sizeX = sizeX;
+		this.sizeY = sizeY;
+		this.hasDisabledState = hasDisabledState;
+		this.type = type;
+	}
+
+	public void setBgTexture(int u0, int v0) {
+		this.u0 = u0;
+		this.v0 = v0;
+	}
+
+	public void setOverlayTexture(int u0, int v0, int size) {
+		setOverlayTexture(u0, v0, size, size);
+	}
+
+	public void setOverlayTexture(int u0, int v0, int sizeX, int sizeY) {
+		this.u1 = u0;
+		this.v1 = v0;
+		this.sizeX = sizeX;
+		this.sizeY = sizeY;
+	}
+
+	public void setZLevel(int z) {
+		this.zLevel = z;
+	}
+
+	@Override
+	public void drawButton(Minecraft mc, int mouseX, int mouseY, float ticks) {
+		if(this.type == EnumButtonType.TAB){
+			if (this.visible) {
+				mc.getTextureManager().bindTexture(GUI_BOOK_TEXTURE);
+				GuiGuideBook.drawStretchingSquare(this.x, this.y, this.x + this.width, this.y + this.height, (int) zLevel, u0, v0, u0 + this.width, v0 + this.height);
+			}
+			if (sizeX > 0 && sizeY > 0)
+				drawTexturedButtonForegroundLayer(mc, mouseX, mouseY, GUI_BOOK_TEXTURE, u1, v1, sizeX, sizeY);
+		}else if(this.type == EnumButtonType.NORMAL){
+			GlStateManager.pushMatrix();
```

Comment:
You're not modifying the transformation matrix, so why push/pop? (In case you are doing that in drawTexturedButtonForegroundLayer, call push/pop in there and not outside)

Reply:
This was before I knew what the transformation matrix stored